internal/datanode/channel_meta.go

```diff
@@ -72,6 +73,18 @@ type Channel interface {
 	RollPKstats(segID UniqueID, stats []*storage.PrimaryKeyStats)
 	getSegmentStatisticsUpdates(segID UniqueID) (*datapb.SegmentStats, error)
 	segmentFlushed(segID UniqueID)
+
+	getChannelCheckpoint(ttPos *internalpb.MsgPosition) *internalpb.MsgPosition
+
+	getCurInsertBuffer(segmentID UniqueID) (*BufferData, bool)
+	setCurInsertBuffer(segmentID UniqueID, buf *BufferData)
```

Comment:
This works for now, but exposing `bufferdata` out side of an interface may cause a concurrency problem. It's much safer to manager `BufferData` inside the ChannelMeta.

No need to change it in this PR, we could refine it later.

Reply:
Got it, I just notify that before this pr, the bufferdata already exposed out, so I didn't change it. And maybe it's ok to keep this, because the flowgraph is a pipeline, no one would manipulate bufferData concurrently, am I right? @XuanYang-cn